### ISSUE TITLE
Custom color scales with rgba values are broken

### ISSUE THREAD

**Issue by tbreloff** [plotly/plotly.js#323]
Adjusting one of your examples:

```
var trace1 = {
  y: [5, 5, 5, 5],
  mode: 'markers',
  marker: {
    size: 40,
    colorscale: [[0,'rgb(255,0.0,0.0)'], [1,'rgba(0.0,0.0,255,0.5)']],
    color: [0, 1, 2, 3]
  }
};

var data = [trace1];

var layout = {
  title: 'Scatter Plot with a Color Dimension'
};

Plotly.newPlot('myDiv', data, layout);
```

The 2nd color of the colorscale is black if I try to use an `rgba`... why?  Is there a way to get this functionality?

**Comment by etpinard** [comment]
Looks like you found a bug. Thanks for reporting :beers: 

With the colorbar on, this issue is even more evident: http://codepen.io/etpinard/pen/zqqeBJ

**Comment by etpinard** [comment]
Here's the reason why: https://github.com/mbostock/d3/issues/582

We have two options:

- write our own interpolate to handle `rgba`  colours
- explicitly not allow `rgba` colours in custom colorscales. The alpha colours would be stripped out in [`is_valiid_scale_array.js`](https://github.com/plotly/plotly.js/blob/master/src/components/colorscale/is_valid_scale_array.js)

**Comment by tbreloff** [comment]
Well I think you know which option I would prefer ;)

The visualization that prompted this is a scatter where the markers get less transparent through time, giving an idea of how the distribution changes.  I currently have to hack around the colorscale rgba limitation by sending a full list of rgba values for every data point.

Here's a toy example in Julia with Plots.jl (plotly backend):

<img width="649" alt="screen shot 2016-03-11 at 6 58 42 pm" src="https://cloud.githubusercontent.com/assets/933338/13718774/524124d6-e7bb-11e5-9379-7506862cd7fc.png">

**Comment by mdtusz** [comment]
Fixed in #422.
